### PR TITLE
Document the various *_accessor typedefs at their source.

### LINKED ISSUE #316: Document iterators at the place where they are typedef'ed
_From [bange...@gmail.com](https://code.google.com/u/111126369971033778052/) on April 09, 2014 12:46:51_

I don't know why this took me 16 years to realize: People always have trouble figuring out which class actually implements the particular iterator they look at (rightfully so, the structure is complicated). There is even a report on just this issue.

What we should be doing is point to the concrete type each iterator represents _at the place where they are typedef'ed_ in the Triangulation and DoFHandler classes.

I don't have the time right now but will get to it before the next release.

_Original issue: http://code.google.com/p/dealii/issues/detail?id=198_

### PR DESCRIPTION
**Pull request by bangerth**
This at long last closes #316. It took only 4.5 years after initially writing
'I don't have the time right now but will get to it before the next release.'

### CODE REVIEW

File: include/deal.II/dofs/dof_handler.h [hunk 1]
```diff
@@ -205,16 +205,100 @@ class DoFHandler : public Subscriptor
...
+   * whereas in three-dimensional meshes the quads are the faces of cells.
+   *
+   * @ingroup Iterators
+   */
+  using quad_iterator = typename ActiveSelector::quad_iterator;
+
+  /**
+   * An alias that allows iterating over the <i>active</i> quads, i.e.,
+   * that subset of quads that have no children. In two-dimensional meshes,
+   * lines these are the cells of the mesh, whereas in three-dimensional
```
Comment (davydden) [inline include/deal.II/dofs/dof_handler.h]: ~~lines~~

File: include/deal.II/grid/tria.h [hunk 2]
```diff
@@ -1580,13 +1580,72 @@ class Triangulation : public Subscriptor
...
+   * whereas in three-dimensional meshes the quads are the faces of cells.
+   *
+   * @ingroup Iterators
+   */
+  using quad_iterator = typename IteratorSelector::quad_iterator;
+
+  /**
+   * An alias that allows iterating over the <i>active</i> quads, i.e.,
+   * that subset of quads that have no children. In two-dimensional meshes,
+   * lines these are the cells of the mesh, whereas in three-dimensional
```
Comment (davydden) [inline include/deal.II/grid/tria.h]: ~~lines~~

File: include/deal.II/hp/dof_handler.h [hunk 3]
```diff
@@ -176,16 +176,100 @@ namespace hp
...
+     * whereas in three-dimensional meshes the quads are the faces of cells.
+     *
+     * @ingroup Iterators
+     */
+    using quad_iterator = typename ActiveSelector::quad_iterator;
+
+    /**
+     * An alias that allows iterating over the <i>active</i> quads, i.e.,
+     * that subset of quads that have no children. In two-dimensional meshes,
+     * lines these are the cells of the mesh, whereas in three-dimensional
```
Comment (davydden) [inline include/deal.II/hp/dof_handler.h]: ~~lines~~

**Comment by drwells** [comment]
Looks good to me. This is just a documentation change so lets merge :)

**Pull request merged**